api/src/services/graphql/resolvers/query.ts

```diff
@@ -39,57 +36,18 @@ export async function resolveQuery(gql: GraphQLService, info: GraphQLResolveInfo
 
 		if (collection.endsWith('_by_version') && collection in gql.schema.collections === false) {
 			collection = collection.slice(0, -11);
-			versionRaw = true;
+			query.versionRaw = true;
 		}
 	}
 
-	if (args['id']) {
-		query.filter = {
-			_and: [
-				query.filter || {},
-				{
-					[gql.schema.collections[collection]!.primary]: {
-						_eq: args['id'],
-					},
-				},
-			],
-		};
-
-		query.limit = 1;
-	}
-
 	// Transform count(a.b.c) into a.b.count(c)
 	if (query.fields?.length) {
 		for (let fieldIndex = 0; fieldIndex < query.fields.length; fieldIndex++) {
 			query.fields[fieldIndex] = parseFilterFunctionPath(query.fields[fieldIndex]!);
 		}
 	}
 
-	const result = await gql.read(collection, query);
-
-	if (args['version']) {
-		const versionsService = new VersionsService({ accountability: gql.accountability, schema: gql.schema });
-
-		const saves = await versionsService.getVersionSaves(args['version'], collection, args['id']);
-
-		if (saves) {
-			if (gql.schema.collections[collection]!.singleton) {
-				return versionRaw
-					? mergeVersionsRaw(result, saves)
-					: mergeVersionsRecursive(result, saves, collection, gql.schema);
-			} else {
-				if (result?.[0] === undefined) return null;
-
-				return versionRaw
-					? mergeVersionsRaw(result[0], saves)
-					: mergeVersionsRecursive(result[0], saves, collection, gql.schema);
-			}
-		}
-	}
-
-	if (args['id']) {
-		return result?.[0] || null;
-	}
+	const result = await gql.read(collection, query, args['id']);
```

Comment:
Any possible side effect due to no longer having this early return? We now process groups every time.